validators/performance/aiperf-bench.Dockerfile

```
@@ -57,8 +57,22 @@ RUN apt-get update \
+# THIRD_PARTY_NOTICES.md are driven by ONE input. tools/generate-python-licenses
+# installs the same file to enumerate what ships here; without a shared input it
+# would have to reconstruct this environment and could silently diverge from it.
+COPY validators/performance/requirements.txt /tmp/requirements.txt
+
+# AIPERF_VERSION now feeds only the io.aicr.aiperf.version label below, so it
+# can drift from what is actually installed. Fail the build closed instead of
+# publishing an image whose label misreports its own contents.
+RUN grep -qE "^aiperf==${AIPERF_VERSION}([[:space:]]|$)" /tmp/requirements.txt \
+ || { echo "ARG AIPERF_VERSION=${AIPERF_VERSION} does not match requirements.txt" >&2; exit 1; }
+
 RUN pip install --no-cache-dir --upgrade pip \
```

Hadolint (2.14.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)